TombEngine/Game/control/box.cpp

```diff
@@ -905,35 +897,26 @@ int ValidBox(ItemInfo* item, short zoneNumber, short boxNumber)
 
 int EscapeBox(ItemInfo* item, ItemInfo* enemy, int boxNumber) 
 {
-	if (boxNumber == NO_BOX)
-		return false;
-
-	auto* box = &g_Level.Boxes[boxNumber];
-
+	auto box = &g_Level.Boxes[boxNumber];
 	int x = (box->top + box->bottom) * SECTOR(1) / 2 - enemy->Pose.Position.x;
 	int z = (box->left + box->right) * SECTOR(1) / 2 - enemy->Pose.Position.z;
-	
-	if (x > -ESCAPE_DIST && x < ESCAPE_DIST && z > -ESCAPE_DIST && z < ESCAPE_DIST)
-		return false;
 
-	if (((x > 0) ^ (item->Pose.Position.x > enemy->Pose.Position.x)) && ((z > 0) ^ (item->Pose.Position.z > enemy->Pose.Position.z)))
+	if (x > -ESCAPE_DIST && x < ESCAPE_DIST &&
+		z > -ESCAPE_DIST && z < ESCAPE_DIST)
+	{
 		return false;
+	}
 
-	return true;
+	return ((z > 0) == (item->Pose.Position.z > enemy->Pose.Position.z)) || ((x > 0) == (item->Pose.Position.x > enemy->Pose.Position.x));
 }
 
 void TargetBox(LOTInfo* LOT, int boxNumber)
 {
-	if (boxNumber == NO_BOX)
-		return;
-
-	boxNumber &= NO_BOX;
-	auto* box = &g_Level.Boxes[boxNumber];
+	auto box = &g_Level.Boxes[boxNumber];
```

Comment:
DO NOT remove asterisks after `auto` if it results in pointer. This is a violation of coding conventions!